gcloud-java-compute/src/main/java/com/google/gcloud/compute/DiskType.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.util.Objects;
+
+/**
+ * A Google Compute Engine disk type. A disk type represents the type of disk to use, such as
+ * {@code pd-ssd} or {@code pd-standard}.
+ *
+ * @see <a href="https://cloud.google.com/compute/docs/reference/latest/diskTypes">Disk Types</a>
+ */
+public final class DiskType implements Serializable {
+
+  static final Function<com.google.api.services.compute.model.DiskType, DiskType> FROM_PB_FUNCTION =
+      new Function<com.google.api.services.compute.model.DiskType, DiskType>() {
+        @Override
+        public DiskType apply(com.google.api.services.compute.model.DiskType pb) {
+          return DiskType.fromPb(pb);
+        }
+      };
+  static final Function<DiskType, com.google.api.services.compute.model.DiskType> TO_PB_FUNCTION =
+      new Function<DiskType, com.google.api.services.compute.model.DiskType>() {
+        @Override
+        public com.google.api.services.compute.model.DiskType apply(DiskType diskType) {
+          return diskType.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = -944042261695072026L;
+
+  private final BigInteger id;
```

Comment:
Here in in other places with "id", do you think there is a value in returning it as a numeric value rather than a String?
I dns (e.g. `ZoneInfo`) we decided to go with String as it feels nicer to something as opaque as id.